dftimewolf/lib/processors/turbinia_gcp.py

```diff
@@ -43,79 +44,69 @@ def __init__(self, state, critical=False):
+    self.run_all_jobs = None
+    self.sketch_id = None
     self.turbinia_region = None
     self.turbinia_zone = None
-    self.sketch_id = None
-    self.run_all_jobs = None
 
   # pylint: disable=arguments-differ
-  def SetUp(self, disk_name, project, turbinia_zone, sketch_id, run_all_jobs):
+  def TurbiniaSetUp(self, project, turbinia_zone, sketch_id, run_all_jobs):
     """Sets up the object attributes.
 
     Args:
-      disk_name (str): name of the disk to process.
       project (str): name of the GPC project containing the disk to process.
       turbinia_zone (str): GCP zone in which the Turbinia server is running.
       sketch_id (int): The Timesketch sketch id
       run_all_jobs (bool): Whether to run all jobs instead of a faster subset.
     """
-    # TODO: Consider the case when multiple disks are provided by the previous
-    # module or by the CLI.
+    self.project = project
+    self.turbinia_zone = turbinia_zone
+    self.sketch_id = sketch_id
+    self.run_all_jobs = run_all_jobs
+
+    turbinia_config.LoadConfig()
+    if not self.project:
+      self.project = turbinia_config.TURBINIA_PROJECT
+    if not self.turbinia_zone:
+      self.turbinia_zone = turbinia_config.TURBINIA_ZONE
 
-    if project is None or turbinia_zone is None:
+    if self.project is None or self.turbinia_zone is None:
       self.state.AddError(
           'project or turbinia_zone are not all specified, bailing out',
           critical=True)
       return
 
-    self.disk_name = disk_name
-    self.project = project
-    self.turbinia_zone = turbinia_zone
-    self.sketch_id = sketch_id
-    self.run_all_jobs = run_all_jobs
-
-    try:
-      turbinia_config.LoadConfig()
-      self.turbinia_region = turbinia_config.TURBINIA_REGION
-      self.instance = turbinia_config.PUBSUB_TOPIC
-      if turbinia_config.TURBINIA_PROJECT != self.project:
-        self.state.AddError(
-            'Specified project {0!s} does not match Turbinia configured '
-            'project {1!s}. Use gcp_turbinia_import recipe to copy the disk '
-            'into the same project.'.format(
-                self.project, turbinia_config.TURBINIA_PROJECT), critical=True)
-        return
-      self._output_path = tempfile.mkdtemp()
-      self.client = turbinia_client.TurbiniaClient()
-    except TurbiniaException as exception:
-      # TODO: determine if exception should be converted into a string as
-      # elsewhere in the codebase.
-      self.state.AddError(exception, critical=True)
+    self.turbinia_region = turbinia_config.TURBINIA_REGION
+    self.instance = turbinia_config.PUBSUB_TOPIC
+    if turbinia_config.TURBINIA_PROJECT != self.project:
+      self.state.AddError(
+          'Specified project {0!s} does not match Turbinia configured '
+          'project {1!s}. Use gcp_turbinia_disk_copy_ts recipe to copy the '
+          'disk into the same project.'.format(
+              self.project, turbinia_config.TURBINIA_PROJECT), critical=True)
```

Comment:
This check becomes moot if we specify that the project is actually the project where turbinia lives. UX-wise, I don't think it makes sense here to let the user pick the project they wanna use (this can be tweaked in the recipes that invoke this module)

Reply:
Yeah, this has actually changed a little bit on the Turbinia side.   By default it should just read the project/zone from the config, and if the values that are passed in to Turbinia are different from the config, it will assume that the disk needs to be copied into the correct project/zone and will copy it over.  This works now because it happens client-side and just uses the same libcloudforensics method that dftimewolf uses.  I'll clean this up a bit on the dftimewolf side to make it more obvious what is going and remove the unneeded params.